tools/swift-format/Sources/Rules/FullyIndirectEnum.swift

```diff
@@ -12,5 +13,67 @@ import SwiftSyntax
 ///
 /// - SeeAlso: https://google.github.io/swift#enum-cases
 public final class FullyIndirectEnum: SyntaxFormatRule {
+  public override func visit(_ node: EnumDeclSyntax) -> DeclSyntax {
+    let enumMembers = node.members.members
+    
+    // Verifies all cases are indirect
+    var it = node.members.members.makeIterator()
+    while let caseMember = it.next()?.decl as? EnumCaseDeclSyntax {
+      guard let caseModifiers = caseMember.modifiers else { return node }
+      if isIndirectCase(modifiers: caseModifiers) { continue }
+      else { return node }
+    }
+    diagnose(.reassignIndirectKeyword(name: node.identifier.text), on: node.identifier)
+    
+    // Removes 'internal' keyword from cases, reformats
+    var newMembers: [MemberDeclListItemSyntax] = []
+    for member in enumMembers {
+      if let caseMember = member.decl as? EnumCaseDeclSyntax {
+        guard let caseModifiers = caseMember.modifiers else { continue }
+        let newCase = caseMember.withModifiers(removeInternalModifier(curModifiers: caseModifiers))
+        let formattedCase = formatCase(unformattedCase: newCase)
+        let newMember = SyntaxFactory.makeMemberDeclListItem(decl: formattedCase, semicolon: nil)
+        newMembers.append(newMember)
+      } else {
+        newMembers.append(member)
+      }
+    }
+
+    let newMemberBlock = SyntaxFactory.makeMemberDeclBlock(leftBrace: SyntaxFactory.makeLeftBraceToken(),
+                                                           members: SyntaxFactory.makeMemberDeclList(newMembers),
+                                                           rightBrace: SyntaxFactory.makeRightBraceToken().withOneLeadingNewline())
+    
+    let newEnumDecl = node.addModifier(SyntaxFactory.makeDeclModifier(name: SyntaxFactory.makeIdentifier("indirect").withOneTrailingSpace(), detailLeftParen: nil, detail: nil, detailRightParen: nil)).withMembers(newMemberBlock)
+    return newEnumDecl
+  }
+  
+  func isIndirectCase(modifiers: ModifierListSyntax) -> Bool {
+        for modifier in modifiers {
+          if modifier.name.tokenKind == .identifier("indirect") {  return true }
+        }
+        return false
+    }
+  
+  func removeInternalModifier(curModifiers: ModifierListSyntax) -> ModifierListSyntax {
+    var newMods: [DeclModifierSyntax] = []
+      for modifier in curModifiers {
+        if modifier.name.tokenKind != .identifier("indirect") { newMods.append(modifier) }
+      }
+    return SyntaxFactory.makeModifierList(newMods)
+  }
+  
+  // Puts new line and indentation in front of the first keyword in the declaration
+  func formatCase(unformattedCase: EnumCaseDeclSyntax) -> EnumCaseDeclSyntax {
+    if let modifiers = unformattedCase.modifiers, let first = modifiers.first {
+      return unformattedCase.withModifiers(modifiers.replacing(childAt: first.indexInParent, with: first.withName(first.name.withLeadingTrivia(.newlines(1) + .spaces(2)))))
```

Comment:
Similar comment from before, you'll want to swap the trivia of the first token in the declaration, instead of hard-coding one newline and two spaces.